File: zeros/mpf.py
import numpy as np

# declaração da função dada
def f(x):
    return x ** 2 - (np.e) ** x

# função de iteração
def phi(x):
    return (-1) * np.sqrt((np.e) ** x)

# chute inicial
initialGuess = -1

# tolerância e máximas iterações
tol = 0.001
maxIter = 10

# método do ponto fixo
def mpf(f, phi, x0, tol, maxIter):

    x_k = [x0]
    k = 0

    while(abs(f(x_k[-1])) > tol and k < maxIter):

        x_next = phi(x_k[-1])
        x_k.append(x_next)
        k += 1

    return x_k[-1]


root = mpf(f, phi, initialGuess, tol, maxIter)

print('Método do ponto fixo: a raiz vale ' + str(root))

File: zeros/mpf.m
format compact;

disp(fixed_point(@f, @phi, -1, 0.0001, 30));

function root = fixed_point(f, phi, x0, tol, max_iter)
    k = 0;
    xi = x0;

    while abs(f(xi)) > tol && k < max_iter
        xi = phi(xi);
        k = k + 1;
    end
    root = xi;
end

function y = f(x)
    y = x^2 - exp(x);
end

function g = phi(x)
    g = (-1) * sqrt(exp(x));
end
